checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -508,6 +540,73 @@ public QualifierHierarchy createQualifierHierarchy(
         return new DeterminismQualifierHierarchy(factory, DET);
     }
 
+    /**
+     * Returns the {@code IdentifierTree} representing the class name of the given tree. For
+     * example, if {@code tree} represents {@code new @NonDet HashSet<String>}, it will return an
+     * identifier representing {@code HashSet}.
+     *
+     * @param tree a tree representing a constructor call
+     * @return an {@code IdentifierTree} representing a class name if it could be found, or {@code
+     *     null} otherwise.
+     */
+    public IdentifierTree getNewClassClassName(NewClassTree tree) {
+        ExpressionTree className = tree.getIdentifier();
+        if (className.getKind() == Tree.Kind.IDENTIFIER) {
+            return (IdentifierTree) className;
+        }
+        if (className.getKind() != Tree.Kind.PARAMETERIZED_TYPE) {
+            return null;
+        }
+        ParameterizedTypeTree paramType = (ParameterizedTypeTree) className;
+        if (paramType.getType().getKind() == Tree.Kind.IDENTIFIER) {
+            return (IdentifierTree) paramType.getType();
+        }
+        if (paramType.getType().getKind() == Tree.Kind.ANNOTATED_TYPE) {
+            AnnotatedTypeTree annoType = (AnnotatedTypeTree) paramType.getType();
+            if (annoType.getUnderlyingType().getKind() == Tree.Kind.IDENTIFIER) {
+                return (IdentifierTree) annoType.getUnderlyingType();
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns the annotation placed on the constructor in {@code tree} if it represents
+     * constructing a parameterized type such as {@code HashSet}.
+     *
+     * @param tree the {@code Tree} representing the construction of a new parameterized type such
+     *     as {@code HashSet}.
+     * @return the annotation in the determinism hierarchy on the given constructor if there was
+     *     one, {@code null} otherwise.
+     */
+    public AnnotationMirror getNewClassAnnotation(NewClassTree tree) {
+        ExpressionTree className = tree.getIdentifier();
+        if (className.getKind() != Tree.Kind.PARAMETERIZED_TYPE) {
+            return null;
+        }
+        ParameterizedTypeTree paramType = (ParameterizedTypeTree) className;
+        if (paramType.getType().getKind() != Tree.Kind.ANNOTATED_TYPE) {
+            return null;
+        }
+        List<? extends AnnotationMirror> annos =
+                TreeUtils.typeOf(paramType.getType()).getAnnotationMirrors();
+        return getQualifierHierarchy().findAnnotationInHierarchy(annos, NONDET);
+    }
+
+    /**
+     * Returns whether type is a variant a {@code @PolyDet}.
+     *
+     * @param the type to test
+     * @return whether type is one of {@code @PolyDet}, {@code @PolyDet("up")},
+     *     {@code @PolyDet("down")}, or {@code @PolyDet("use")}.
+     */
+    private boolean isPolyDetVariant(AnnotationMirror type) {
```

Comment:
As we discussed earlier today, this can be replaced with AnnotationUtils.areSameIgnoringValues()

Reply:
That method has been recently renamed to `areSameByName`.  Please pull from master and use the new name.